packages/ui/src/2_molecules/Dialog/Dialog.tsx

```diff
@@ -44,6 +47,35 @@ export const Dialog: IDialogFunctionComponent<DialogProps> = ({
     Dialog.index++;
   }, []);
 
+  useEffect(() => {
+    const handleEscape = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        event.stopPropagation();
+        // close the dialog only if it's the topmost one
+        if (
+          ref.current === document.activeElement ||
+          ref.current?.contains(document.activeElement)
+        ) {
+          handleClose();
+        }
+      }
+    };
+
+    if (closeOnEscape) {
+      document.addEventListener('keydown', handleEscape);
+    }
+
+    return () => {
+      document.removeEventListener('keydown', handleEscape);
+    };
+  }, [handleClose, closeOnEscape]);
+
+  useEffect(() => {
+    if (isOpen && initialFocusRef && initialFocusRef.current) {
+      initialFocusRef.current.focus();
+    }
+  }, [isOpen]);
```

Comment:
```suggestion
  }, [initialFocusRef, isOpen]);
```

Reply:
Normally  adding `initialFocusRef` as dependency would be unnecessary because it's a ref and is mutable value. 
Do behaviour changes here because it's passed as a prop?

Reply:
I don't know but my linter is screaming at me :D

Reply:
will make some testing, docs doesn't say anything about refs as prop :)

Reply:
Ok, including it doesn't break anything so added as dependency

Reply:
@creed-victor btw, if you use a `ref` as a `prop`, it acts as a normal property. That's one disadvantage of this approach. If you want, you can use `forwardRef` and it will work as you'd expect (but it's not necessary in this case).